buildSrc/src/main/groovy/airbyte-docker.gradle

```diff
@@ -0,0 +1,216 @@
+
+    @Input
+    String testDockerfileName = 'Dockerfile.test'
+
+    @OutputFile
+    abstract File idFileOutput
+
+    def buildDockerfile(String scriptPath, String fileName) {
+        if(project.file(fileName).exists()) {
+            def tag = getTaggedImage(projectDir, dockerfileName)
+
+            project.exec {
+                commandLine scriptPath, rootDir.absolutePath, projectDir.absolutePath, dockerfileName, tag, idFileOutput.absolutePath
+            }
+        }
+    }
+
+    @TaskAction
+    def airbyteDocker() {
+        def scriptPath = Paths.get(rootDir.absolutePath, 'tools/bin/build_image.sh').toString()
+        buildDockerfile(scriptPath, dockerfileName)
+        buildDockerfile(scriptPath, testDockerfileName)
+    }
+
+    static String extractLabelValue(String dockerFile, String labelName) {
+        def file = dockerFile instanceof File ? dockerFile : new File(dockerFile)
+        return file.readLines()
+                .grep({ it.startsWith('LABEL') && it.contains(labelName) })
+                .get(0)
+                .split('=')[1]
+    }
+
+    static String extractImageName(String dockerFile) {
+        return extractLabelValue(dockerFile, "io.airbyte.name")
+    }
+
+    static String extractImageVersion(String dockerFile) {
+        return extractLabelValue(dockerFile, "io.airbyte.version")
+    }
+
+    static String getTaggedImage(projectDir, dockerfileName) {
+        return "${extractImageName(Paths.get(projectDir.absolutePath, dockerfileName).toString())}:dev"
+    }
+}
+
+class AirbyteDockerPlugin implements Plugin<Project> {
+
+    static def getBaseTaggedImages(File dockerfile) {
+        def result = [] as Set<String>
+
+        dockerfile.eachLine { line ->
+            if (line.startsWith("FROM ")) {
+                def image = line.split()[1]
+                result.add(image)
+            } else if(line.startsWith("COPY --from=")) {
+                def image = line.substring("COPY --from=".length()).split()[0]
+                result.add(image)
+            }
+        }
+
+        return result
+    }
+
+    static def getBaseImageHashes(Map imageToVersion, File dockerfile) {
+        def result = [] as Set<String>
+
+        getBaseTaggedImages(dockerfile).forEach { taggedImage ->
+            result.add((String)imageToVersion.get(taggedImage))
+        }
+
+        return result
+    }
+
+    static String getImageHash(Project project, String taggedImage) {
+        def stdout = new ByteArrayOutputStream()
+
+        project.exec {
+            commandLine "docker", "images", "--no-trunc", "-f", "dangling=false", "--format", "{{.ID}}", taggedImage
+            standardOutput = stdout;
+        }
+
+       return "$stdout".toString().trim()
+    }
+
+    static boolean isUpToDate(Logger logger, File idFileOutput, Project project, Path dockerPath) {
+        if(idFileOutput.exists()) {
+            def taggedImage = AirbyteDockerTask.getTaggedImage(project.projectDir, 'Dockerfile')
```

Comment:
should the dockerfile name be hardcoded?

Reply:
fixed